src/app/courses/components/course/course.component.ts

```diff
@@ -5,10 +5,13 @@ import { Course } from '../../../shared/interfaces/course';
   selector: 'app-course',
   templateUrl: './course.component.html',
   styleUrls: ['./course.component.scss'],
-  changeDetection:  ChangeDetectionStrategy.OnPush,
+  // changeDetection:  ChangeDetectionStrategy.OnPush, <- break update data after save on server
 })
 export class CourseComponent {
   @Input() course:  Course;
+  @Input() canEdit: boolean;
```

Comment:
It's a good approach to set default value here, to avoid a lot of information to set as input for this component.